### ISSUE TITLE
[Accessibility]: CodeMirror suggestions read all in one go and icons not read

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43139]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188

### Problem description

CodeMirror suggestions are read all in one go, with no pause between the filter and the description. Also, the icons are not read; this is not a problem for filter suggestions as the type is in the description, but filter and repo suggestions don't 

<img width="739" alt="image" src="https://user-images.githubusercontent.com/206864/196557638-0ec4209b-fdfe-4a4c-ba46-63bd7c9173a6.png">

### Expected behavior

- There should be some kind of pause between the name and the description (this can be achieved with a `<span className="sr-only">,</span>` I think)
- Non-symbol suggestions should have a type read (as represented by the icon; probably should use an `aria-label` on the icon). May want to do this for symbol suggestions too for consistency.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @lguychard

**Comment by fkling** [comment]
It would be nice if we could add `aria-label`s but it's not possible to modify the DOM elements that CodeMirror creates for the suggestions. But we can insert new DOM elements in-between every element, so adding `,` should be no problem.
